### PR TITLE
KAFKA-15521: Refactor build.gradle to align gradle swagger plugin with swagger dependencies

### PR DESCRIPTION
**Pull request by atu-sharm**
Having a single version of swagger to avoid breaking of build

*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by atu-sharm** [comment]
As per 

- Gradle docs : https://docs.gradle.org/current/userguide/plugins.html#sec:plugin_version_management:~:text=will%20be%20checked.-,Plugin%20Version%20Management,-A%20plugins%20%7B%7D

- PR: https://github.com/gradle/gradle/issues/1697#issuecomment-506910915 

"We cannot call methods or use sophisticated build logic to retrieve dependency versions. But anything declared in a gradle.properties file can be inserted into the version string."

Made changes accordingly

**Comment by atu-sharm** [comment]
Hey @mimaison can you please review this?

File: gradle/dependencies.gradle [hunk 1]
```diff
@@ -158,9 +158,6 @@ versions += [
   slf4j: "1.7.36",
   snappy: "1.1.10.5",
   spotbugs: "4.7.3",
-  // New version of Swagger 2.2.14 requires minimum JDK 11.
```
Comment (mimaison) [inline gradle/dependencies.gradle:161]: Can we keep this comment in gradle.properties?
I created https://issues.apache.org/jira/browse/KAFKA-15549 to remember us we will be able to bump it once support for Java 8 is removed.

File: build.gradle [hunk 2]
```diff
@@ -938,7 +938,7 @@ project(':core') {
       exclude group: 'junit', module: 'junit'
     }
     testImplementation libs.caffeine
-
```
Comment (mimaison) [inline build.gradle]: Can we undo this change?

File: build.gradle [hunk 3]
```diff
@@ -37,7 +37,7 @@ plugins {
   id 'java-library'
   id 'org.owasp.dependencycheck' version '8.2.1'
   id 'org.nosphere.apache.rat' version "0.8.1"
-  id "io.swagger.core.v3.swagger-gradle-plugin" version "2.2.8" // keep aligned with the swagger version used in gradle/dependencies.gradle
+  id "io.swagger.core.v3.swagger-gradle-plugin" version "${swaggerVersion}" // keep aligned with the swagger version used in gradle/dependencies.gradle
```
Comment (mimaison) [inline build.gradle]: I guess we can remove the comment if we have the version in a single place

**Review by mimaison — COMMENTED** [review]
Thanks for the PR. I left a few comments

**Comment by atu-sharm** [comment]
Thanks, @mimaison for the review, added changes.

**Review by mimaison — APPROVED** [review]
LGTM, thanks for the PR

File: gradle.properties [hunk 4]
```diff
@@ -25,6 +25,9 @@ group=org.apache.kafka
 #  - streams/quickstart/java/pom.xml
 version=3.7.0-SNAPSHOT
 scalaVersion=2.13.12
+# Adding swaggerVersion in gradle.properties to have a single version in place for swagger
+# New version of Swagger 2.2.14 requires minimum JDK 11.
+swaggerVersion=2.2.8
```
Comment (ijuma) [inline gradle.properties:30]: Why didn't we keep this in `dependencies.gradle`? It's confusing to have versions split across multiple files (outside of the Scala version, which is special in a few ways).

Comment (atu-sharm) [reply gradle.properties:30]: Yes, i agree @ijuma , but as per the https://github.com/gradle/gradle/issues/1697#issuecomment-506910915 , plugin block is highly constrained, as

1. Only the version accepts interpolated strings (id must be a string literal)
2. Replacement expressions are limited to variable references: this means that we cannot call methods or use sophisticated build logic to retrieve dependency versions. But anything declared in a gradle.properties file can be inserted into the version string

So, i made changes accordingly.

Apparently we can have another approach possible where we can add specific resolutionStrategy using the pluginManagement block in settings.gradle  and specify to use the specific plugin having the version defined in dependencies.gradle, this would work as well.

Your thoughts @ijuma
